### PR TITLE
P4-2392 intial piece of work for integrating with the HMPPS auth service.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
This is start of integration with the HMPPS auth service.

The localstack changes has a dependency on this PR also [PECS jpc test creds](https://github.com/ministryofjustice/hmpps-auth/pull/616)

### CODE REVIEW

File: helm_deploy/calculate-journey-variable-payments/templates/secrets.yaml [hunk 1]
```diff
@@ -7,4 +7,6 @@ data:
   BASM_BUCKET_NAME: {{ .Values.secrets.BASM_BUCKET_NAME | b64enc | quote }}
   BASM_AWS_ACCESS_KEY_ID: {{ .Values.secrets.BASM_AWS_ACCESS_KEY_ID | b64enc | quote }}
   BASM_AWS_SECRET_ACCESS_KEY: {{ .Values.secrets.BASM_AWS_SECRET_ACCESS_KEY | b64enc | quote }}
+  HMPPS_AUTH_CLIENT_ID: {{ .Values.secrets.HMPPS_AUTH_CLIENT_ID | b64enc | quote }}
```
Comment (matt-crowson-moj) [inline helm_deploy/calculate-journey-variable-payments/templates/secrets.yaml:10]: @mafonso helm linter doesn't like this

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/config/SecurityConfiguration.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package uk.gov.justice.digital.hmpps.pecs.jpc.config
+
+import org.springframework.context.annotation.Profile
+import org.springframework.security.config.annotation.web.builders.HttpSecurity
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
+import org.springframework.security.config.web.servlet.invoke
+
+// TODO for now this is tied to run for localstack profile only!!
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/config/SecurityConfiguration.kt:9]: FYI @jcavell

File: src/main/resources/application-localstack.yml [hunk 3]
```diff
@@ -18,3 +18,28 @@ import-files:
...
   template: classpath:/spreadsheets/JPC_template.xlsx
+
+spring:
+  security:
+    oauth2:
+      client:
+        provider:
+          hmpps:
+            authorization-uri: http://localhost:9090/auth/oauth/authorize
+#            authorization-uri: ${hmpps.auth.base.uri}/oauth/authorize
```
Comment (matt-crowson-moj) [inline src/main/resources/application-localstack.yml:29]: The commented out lines is what it should be when not limited to localstack

File: src/main/resources/application-localstack.yml [hunk 4]
```diff
@@ -18,3 +18,28 @@ import-files:
...
+        registration:
+          hmpps:
+            client-id: pecs-jpc-client
+#            client-id: ${hmpps.auth.client.id}
+            client-secret: clientsecret
+#            client-secret: ${hmpps.auth.client.secret}
+            provider: hmpps
+            scope: read
+            authorization-grant-type: authorization_code
+            redirect-uri: http://localhost:8080/login/oauth2/code/hmpps
```
Comment (matt-crowson-moj) [inline src/main/resources/application-localstack.yml:44]: This uri is set up in hmpps-auth and it calls back to this.  The format is specific to Spring.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/ImportControllerTest.kt [hunk 5]
```diff
@@ -28,6 +32,7 @@ import java.time.LocalDateTime
 @TestInstance(TestInstance.Lifecycle.PER_CLASS)
 @ActiveProfiles("test")
 @ContextConfiguration(classes = [TestConfig::class])
+@Disabled
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/ImportControllerTest.kt:35]: Not ideal.  Realised at end of day that need to get this working as it is picking up the security config, grrr.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/config/SecurityConfiguration.kt [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    @Throws(Exception::class)
+    override fun configure(http: HttpSecurity) {
+        http {
+            authorizeRequests {
+                authorize(anyRequest, authenticated)
+            }
+            exceptionHandling {
+            }
+            oauth2Login {
+                defaultSuccessUrl("/", true)
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/config/SecurityConfiguration.kt:23]: This needs to go to a page when we have one in place (choose supplier?? by default when available).

**Review by jcavell — APPROVED** [review]

**Pull request merged**